component/coredns.libsonnet

```diff
@@ -0,0 +1,238 @@
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+local inv = kap.inventory();
+local common = import 'common.libsonnet';
+// The hiera parameters for the component
+local params = inv.parameters.vcluster;
+
+local corednsConfigMap =
+  function(name, namespace)
+    kube.ConfigMap('vc-%s-coredns' % name) {
+      metadata+: {
+        namespace: namespace,
+      },
+      data: {
+        // The deployment has some variables in there that get modified by vcluster.
+        // It is not valid yaml, so we need to use a string.
+        // The Helm chart does use a string too.
```

Comment:
I don't remember if we tried using the Helm chart in the component previously, but I wonder if we could actually reduce the amount of static template code in the component significantly by using it.

Reply:
I think back then we had some troubles since we could not render the Helm chart with a values.yml from jsonnet. But I also don't remember fully why this was an issue.

Reply:
Might be worth to create a follow-up issue to look at changing the component to be based on the Helm chart.